File: sorting/selection/selection.py
def selection(arr):

    for i in range(0, len(arr)):
        # Min idx starts always at the ith element
        min_idx = i

        # Look for any local minima going forward
        for j in range(i+1, len(arr)):
            
            # If we find one, set the min_idx to what we find
            if arr[j] < arr[min_idx]:
                min_idx = j

        # Swap that min with the location at i
        arr[i], arr[min_idx] = arr[min_idx], arr[i]

if __name__ == '__main__':
    arrs = [
        [4, 3, 3, 7, 6, -1, 10, 3, 8, 4],
        [10, 11, 9, 8, 13, 21],
        [-1, -1]
    ]
    print("Selection sort")
    for arr in arrs:
        print("----------------------------------------")
        print("unsorted: {}".format(arr))
        selection(arr)
        print("sorted: {}".format(arr))

    print("\n")
    print("Selection sort")
    for arr in arrs:
        print("----------------------------------------")
        print("unsorted: {}".format(arr))
        selection(arr)
        print("sorted: {}".format(arr))

File: sort/selection/selection.js
"use strict";

function selection(arr) {
    for (let i = 0; i < arr.length; i++) {
        let min = i;
        for (let j = i + 1; j < arr.length; j++) {
            if (arr[j] < arr[min]) {
                min = j;
            }
        }
        let temp = arr[i];
        arr[i] = arr[min];
        arr[min] = temp;
    }
}

if (require.main === module) {
    console.log("Selection sort");
    let arr = [8, 9, 7, 5, 2, 4, 6, 1, 3];
    selection(arr);
    console.log(arr);
}
